drivers/mipi_dsi/dsi_stm32.c

```diff
@@ -0,0 +1,465 @@
+			return -ret;
+		}
+	}
+
+	return 0;
+}
+
+static ssize_t mipi_dsi_stm32_transfer(const struct device *dev, uint8_t channel,
+				       struct mipi_dsi_msg *msg)
+{
+	struct mipi_dsi_stm32_data *data = dev->data;
+	uint32_t param1 = 0;
+	uint32_t param2 = 0;
+	ssize_t len;
+	int ret;
+
+	switch (msg->type) {
+	case MIPI_DSI_DCS_READ:
+		ret = HAL_DSI_Read(&data->hdsi, channel, (uint8_t *)msg->rx_buf, msg->rx_len,
+				   msg->type, msg->cmd, (uint8_t *)msg->rx_buf);
+		len =  msg->rx_len;
+		break;
+	case MIPI_DSI_DCS_SHORT_WRITE:
+	case MIPI_DSI_DCS_SHORT_WRITE_PARAM:
+		param1 = msg->cmd;
+		if (msg->tx_len >= 1U) {
+			param2 = ((uint8_t *)msg->tx_buf)[0];
+		}
+
+		ret = HAL_DSI_ShortWrite(&data->hdsi, channel, msg->type, param1, param2);
+		len = msg->tx_len;
+		break;
+	case MIPI_DSI_DCS_LONG_WRITE:
+		ret = HAL_DSI_LongWrite(&data->hdsi, channel, msg->type, msg->tx_len, msg->cmd,
+					(uint8_t *)msg->tx_buf);
+		len = msg->tx_len;
+		break;
+	case MIPI_DSI_GENERIC_SHORT_WRITE_0_PARAM:
+	case MIPI_DSI_GENERIC_SHORT_WRITE_1_PARAM:
+	case MIPI_DSI_GENERIC_SHORT_WRITE_2_PARAM:
+		param1 = ((uint8_t *)msg->tx_buf)[0];
+		if (msg->tx_len == 1U) {
+			param2 = ((uint8_t *)msg->tx_buf)[1];
+		}
+
+		if (msg->tx_len >= 2U) {
+			param2 = *(uint16_t *)&((uint8_t *)msg->tx_buf)[1];
+		}
+
+		ret = HAL_DSI_ShortWrite(&data->hdsi, channel, msg->type, param1, param2);
+		len = msg->tx_len;
+		break;
+	case MIPI_DSI_GENERIC_LONG_WRITE:
+		ret = HAL_DSI_LongWrite(&data->hdsi, channel, msg->type, msg->tx_len,
+					((uint8_t *)msg->tx_buf)[0], &((uint8_t *)msg->tx_buf)[1]);
+		len = msg->tx_len;
+		break;
+	default:
+		LOG_ERR("Unsupported message type (%d)", msg->type);
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_MIPI_DSI_LOG_LEVEL_DBG)) {
+		char tmp[64];
+
+		if (msg->type == MIPI_DSI_DCS_READ) {
+			snprintk(tmp, sizeof(tmp), "RX: ch %3d, reg 0x%02x, len %2d",
+				 channel, msg->cmd, msg->rx_len);
+			LOG_HEXDUMP_DBG(msg->rx_buf, msg->rx_len, tmp);
+		} else {
+			snprintk(tmp, sizeof(tmp), "TX: ch %3d, reg 0x%02x, len %2d",
+				 channel, msg->cmd, msg->tx_len);
+			LOG_HEXDUMP_DBG(msg->tx_buf, msg->tx_len, tmp);
+		}
+	}
+
+	if (ret != HAL_OK) {
+		LOG_ERR("Transfer failed! (%d)", ret);
+		return -EIO;
+	}
+
+	return len;
+}
+
+static struct mipi_dsi_driver_api dsi_stm32_api = {
+	.attach = mipi_dsi_stm32_attach,
+	.transfer = mipi_dsi_stm32_transfer,
+};
+
+static int mipi_dsi_stm32_init(const struct device *dev)
+{
+	const struct mipi_dsi_stm32_config *config = dev->config;
+	int ret;
+
+	if (!device_is_ready(config->rcc)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(config->rcc, (clock_control_subsys_t)&config->pclken);
+	if (ret < 0) {
+		LOG_ERR("Enable DSI peripheral clock failed! (%d)", ret);
+		return ret;
+	}
+
+	/** Soft Reset the DSI Host and wrapper */
+	__HAL_RCC_DSI_FORCE_RESET();
+	__HAL_RCC_DSI_RELEASE_RESET();
```

Comment:
ST needs to implement a proper reset driver, same as it was done for GD32, cc: @erwango

Reply:
Done ;-) :
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/reset/reset_stm32.c